crates/miden-objects/src/block/note_tree.rs

```diff
@@ -104,15 +104,17 @@ pub struct BlockNoteIndex {
 
 impl BlockNoteIndex {
     /// Creates a new [BlockNoteIndex].
-    pub fn new(batch_idx: usize, note_idx_in_batch: usize) -> Result<Self, BlockError> {
-        if note_idx_in_batch >= MAX_OUTPUT_NOTES_PER_BATCH {
-            return Err(BlockError::TooManyNotesInBatch(note_idx_in_batch));
-        }
-        if batch_idx >= MAX_BATCHES_PER_BLOCK {
-            return Err(BlockError::TooManyTransactionBatches(batch_idx));
-        }
-
-        Ok(Self { batch_idx, note_idx_in_batch })
+    ///
+    /// # Panics
+    ///
+    /// Panics in debug mode if the batch index exceeds is equal to or greater than
+    /// [`MAX_BATCHES_PER_BLOCK`] or if the note index is equal to or greater than
+    /// [`MAX_OUTPUT_NOTES_PER_BATCH`].
+    pub fn new(batch_idx: usize, note_idx_in_batch: usize) -> Self {
+        debug_assert!(note_idx_in_batch < MAX_OUTPUT_NOTES_PER_BATCH);
+        debug_assert!(batch_idx < MAX_BATCHES_PER_BLOCK);
```

Comment:
Should we change these to regular `assert!` since this type is exported publicly from the library?